moler/connection_observer.py

```diff
@@ -46,6 +46,9 @@ def __init__(self, connection=None, runner=None):
         self._exception = None
         self.runner = runner if runner else get_runner()
         self._future = None
+        self.inactivity_timeout = 0.0  # If positive value and no data are sent by connection in this time then method
+        #                                on_inactivity will be called.
+        self.last_feed_time = None  # Time of last called data_received or on_inactivity.
```

Comment:
change it to `_last_feed_time` since it is used only internally by runner

change comment to: `# Last time data_received() or on_inactivity() was called (for runner internal use)`

Reply:
More generic question:

Shouldn't we refactor all those parameters of connection_observer that are used only internally by runner (for tracking lifetime progress of connection_observer)?

How about putting all such attributes under single class:
```python
class LifeStatus(object):
    def __init__(self):
        self.start_time = 0.0
        self.last_feed_time = None
        self.in_terminating = False
        self.was_on_timeout_called = False
```
as you can see I've put `self.start_time` here as well even if it is set by connection_observer and not runner. But all those parameters are related to lifetime progress.

and then use it:
```python
class ConnectionObserver(object):
    def __init__(self, connection=None, runner=None):
        self._life_status = LifeStatus()
```

I opt for such solution since it simplifies maintenance:
1. all lifetime related attribs are in single place
1. adding new ones doesn't need modification inside connection_observer.py

Reply:
Done